### ISSUE TITLE
Adding Gatsby blog to existing site

### ISSUE THREAD

**Issue by pkhivesara** [gatsbyjs/gatsby#16664]
## Summary

I am trying to add a Gatsby blog to an existing site. However, I keep getting a 'broken link' error in production.

Here are the steps that I have followed (I might be terribly off base):

Created a subdirectory in existing website. For example: myawesomesite.com/blog

Added the Gatsby blog template within the above folder.

Also changed the gatsby-config.js siteURL to point to myawesomesite.com/blog.

I am a completely noob with Gatsby.

The blog template works fine on its own however it fails to run when added as a subdirectory to the existing site. My current site is based on Bootstrap 4. I am essentially trying to have a menu item that when clicked takes the user to a new page (i.e. blog) which is built on gatsby. Is is possible to do so?

Thanks!

### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

**Comment by matowang** [comment]
After you build your gatsby site using `gatsby build` in the terminal you will see a public folder with files in it. Something like this:

/blog-webiste
--/public
-----index.html
-----component1.js
-----foo1
-----foo2

Those files are the files you want to put in your website. Just put it in the /blog directory

/your-website
--index.html
--styles.css
--/blog
-----index.html
-----component1.js
-----foo1
-----foo2

Now going to myawesomesite.com/blog will lead you to the /blog/index.html created in Gatsby. This is not Gatsby specific. That's just how basic  html routing works.
Note: gatsby-config.js siteMetaData has nothing to do with any real settings. It's just there for you to query some information.

**Comment by pkhivesara** [comment]
@matowang  Thanks for your response! I tried the approach you suggested, however when I run the site and try to go to /blog/index.html, it shows me a list of the directories instead of going to index.html.
My href in the website points to blog/.

<img width="748" alt="Screen Shot 2019-08-16 at 7 09 26 AM" src="https://user-images.githubusercontent.com/6237488/63163966-e7006900-bff4-11e9-8246-878524af73af.png">

What am I missing here?
Thanks!

**Comment by m-allanson** [comment]
Hey @pkhivesara to serve your Gatsby site from a subdirectory like `/blog` you'll need to set the Gatsby's `pathPrefix` option in your gatsby-config.js. [See here for docs on how to do that](https://www.gatsbyjs.org/docs/path-prefix/).

Once that's done, you'll need to run `gatsby build --prefix-paths`, then follow the steps described by @matowang.

**Comment by pkhivesara** [comment]
Hey @m-allanson  So I was able to get the blog to run using the approach as described by @matowang .
However the approach only shows the list of blog posts. If I click on any one of them the URL changes from awesomesite/blog/mypost to awesomesite/mypost.

I have also noticed that favicon is not visible on the list of blog posts.

I tried the pathPrefix approach as you mentioned but that's not working either. Maybe I am missing something?
Thanks!
